flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/NoFetchingInput.java

```diff
@@ -65,6 +65,10 @@ public boolean canReadLong() throws KryoException {
      */
     @Override
     protected int require(int required) throws KryoException {
+        if (required == 0) {
+            return 0;
+        }
```

Comment:
Hi @qinghui-xu , I'm not sure could this change fix the bug that FLINK-34954 mentioned. Would you mind helping double check? Thank you in advance.